control-plane/connect-inject/controllers/endpoints/endpoints_controller_test.go

```diff
@@ -715,6 +715,7 @@ func TestReconcileCreateEndpoint_MultiportService(t *testing.T) {
 						constants.MetaKeyKubeNS:  "default",
```

Comment:
Should we add a case for TestReconcileUpdateEndpoint or TestReconcileDeleteEndpoint or even a new test case that's a bit more end to end that ensures that with a non-empty pod uid meta, only acl token with matching pod uid is deleted?

Reply:
updated the TestReconcileDeleteEndpoint to support deletion using pod uid